### PR TITLE
refactor cycleData/originalDataPoint endpoint

### PR DESCRIPTION
**Pull request by sorja**
part of #1458

### CODE REVIEW

File: meta/api/endpoint/ApiEndPoint.ts [hunk 1]
```diff
@@ -19,6 +19,17 @@ export const ApiEndPoint = {
 
   CycleData: {
     descriptions: () => apiPath('cycleData', 'descriptions'),
+
+    OriginalDataPoint: {
+      one: () => apiPath('cycleData', 'originalDataPoints', 'originalDataPoint'),
+
+      // Used for print view
+      many: () => apiPath('cycleData', 'originalDataPoints'),
```
Comment (sorja) [inline meta/api/endpoint/ApiEndPoint.ts:27]: printView fetches many originalDataPoints (all infos) and displays them before ODP tables

File: meta/api/endpoint/ApiEndPoint.ts [hunk 2]
```diff
@@ -19,6 +19,17 @@ export const ApiEndPoint = {
...
+
+      // Table Data (1a, 1b)
+      data: () => apiPath('cycleData', 'originalDataPoints', 'data'),
```
Comment (sorja) [inline meta/api/endpoint/ApiEndPoint.ts:30]: where as `data` is only used to mix the data with the tables (ex. 1a, 1b)

File: server/api/cycleData/index.ts [hunk 3]
```diff
@@ -23,6 +31,19 @@ export const CycleDataApi = {
...
+    express.post(ApiEndPoint.CycleData.OriginalDataPoint.one(), AuthMiddleware.requireEdit, createOriginalDataPoint)
+    express.delete(ApiEndPoint.CycleData.OriginalDataPoint.one(), AuthMiddleware.requireEdit, deleteOriginalDataPoint)
+    express.get(ApiEndPoint.CycleData.OriginalDataPoint.one(), AuthMiddleware.requireView, getOriginalDataPoint)
+    express.put(ApiEndPoint.CycleData.OriginalDataPoint.one(), AuthMiddleware.requireEdit, updateOriginalDataPoint)
+
+    express.get(ApiEndPoint.CycleData.OriginalDataPoint.many(), AuthMiddleware.requireView, getOriginalDataPoints)
+
+    // OriginalDataPoint // table
+    express.get(ApiEndPoint.CycleData.OriginalDataPoint.data(), AuthMiddleware.requireView, getOriginalDataPointData)
+
```
Comment (sorja) [inline server/api/cycleData/index.ts:46]: should we add index.ts with `export OriginalDataPointApi (express) => ..` and replace this with `OriginalDataPointApi.init(express)` ?

Comment (rfontanarosa) [reply server/api/cycleData/index.ts:46]: I think that this way is fine

**Review by rfontanarosa — APPROVED** [review]

**Review by minotogna — APPROVED** [review]

**Pull request merged**